pkg/state/state_ofed.go

```diff
@@ -464,6 +476,56 @@ func (s *stateOFED) getManifestObjects(
 	return objs, nil
 }
 
+// add pull secrets from the init container to the pull secrets list of the main container,
+// this list will be used as imagePullSecrets on the Pod level
+func (s *stateOFED) mergeImagePullSecrets(cr *mellanoxv1alpha1.NicClusterPolicy) {
+	if cr.Spec.OFEDDriver.InitContainer == nil {
+		return
+	}
+	if len(cr.Spec.OFEDDriver.InitContainer.ImagePullSecrets) == 0 {
+		return
+	}
+	secretsToAdd := make([]string, 0, len(cr.Spec.OFEDDriver.InitContainer.ImagePullSecrets))
+	for _, initSecret := range cr.Spec.OFEDDriver.InitContainer.ImagePullSecrets {
+		found := false
+		for _, mainSecret := range cr.Spec.OFEDDriver.ImagePullSecrets {
+			if initSecret == mainSecret {
+				found = true
+				break
+			}
+		}
+		if !found {
+			secretsToAdd = append(secretsToAdd, initSecret)
+		}
+	}
+	cr.Spec.OFEDDriver.ImagePullSecrets = append(cr.Spec.OFEDDriver.ImagePullSecrets, secretsToAdd...)
+}
+
+// prepare configuration for the init container
+func (s *stateOFED) getInitContainerConfig(
+	cr *mellanoxv1alpha1.NicClusterPolicy, reqLogger logr.Logger) initContainerConfig {
+	var initContCfg initContainerConfig
+	safeLoadEnable := cr.Spec.OFEDDriver.OfedUpgradePolicy != nil &&
+		cr.Spec.OFEDDriver.OfedUpgradePolicy.AutoUpgrade &&
+		cr.Spec.OFEDDriver.OfedUpgradePolicy.SafeLoad
```

Comment:
what will happen if:

1. init container is present
2. safe load is disabled
3. mofed is ready
4. user enables safe load 

what we want: nothing happens in cluster except config map update
what i think will happen: reload of mofed in entire cluster